src/userscores/qml/MuseScore/UserScores/ExportDialog.qml

```diff
@@ -0,0 +1,237 @@
+
+                                    height: 30
+                                    width: rightRectangle.width * 0.8
+
+                                    currentText: scoresModel.getExportPath()
+
+                                    onCurrentTextEdited: {
+                                        scoresModel.setExportPath(newTextValue);
+                                    }
+                                }
+
+                                FlatButton {
+                                    id: browsePathButton
+
+                                    icon: IconCode.NEW_FILE
+
+                                    height:30
+                                    width: rightRectangle.width * 0.1
+
+                                    onClicked: {
+                                        fileExportPathInput.currentText = scoresModel.chooseExportPath();
+                                    }
+                                }
+
+                            }
+                        }
+
+
+                        Column {
+                            Layout.fillWidth: true
+                            Layout.preferredHeight: childrenRect.height
+
+                            spacing: 6
+
+                            StyledTextLabel {
+                                Layout.fillWidth: true
+                                Layout.topMargin: 24
+
+                                text: qsTrc("userscores", "Export As: ")
+                                font.capitalization: Font.AllUppercase
+
+                            }
+
+                            StyledComboBox {
+                                id: fileTypeSelect
+
+                                property var currValue: 1
+
+                                width: rightRectangle.width * 0.8
+
+                                textRoleName: "text"
+                                valueRoleName: "value"
+
+                                model: [
+                                    { text: "PDF File", value: 1, suffix: "pdf" },
+                                    { text: "PNG Image", value: 2, suffix: "png" },
+                                    { text: "SVG Image", value: 3, suffix: "svg" },
+                                    { text: "MIDI", value: 4, suffix: "midi" },
+                                    { text: "mp3", value: 5, suffix: "mp3" },
+                                    { text: "ogg", value: 6, suffix: "ogg" },
+                                ]
+
+                                currentIndex: indexOfValue(currValue)
+
+                                onValueChanged: {
+                                    currValue = value;
+                                    scoresModel.setExportSuffix(valueFromModel(indexOfValue(value), "suffix"));
+                                    fileExportPathInput.currentText = scoresModel.getExportPath();
+                                }
+                            }
+                        }
+
+
+                        ExportFilePages {
```

Comment:
Each writer has its options. This is now hardcoded into qml. 
This can be made universal, as in the settings(see SettingsPage.qml). As in MU3. When a new writer is added, the options will be showed on the page automatically.

Reply:
I've gotten this to work using the settings page as a base. But how do i go about getting the additional info needed for each input, things like min/max values, suffixes for the number inputs or user friendly names for the settings keys?